src/main/scala/decider/TermToZ3APIConverter.scala

```diff
@@ -30,6 +30,7 @@ class TermToZ3APIConverter
 
   val sortCache = mutable.HashMap[Sort, Z3Sort]()
   val funcDeclCache = mutable.HashMap[(String, Seq[Sort], Sort), Z3FuncDecl]()
+  val termCache = mutable.HashMap[Term, Z3Expr]()
```

Comment:
I guess we don't need a thread-safe collection here because the term-layer below, where terms are constructed, uses a thread-safe one? That's nice!

Reply:
Exactly. And we also don't need an IdentityHashmap here because the term layer already uses identity for equality ^^